plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -945,7 +961,11 @@ class AzBatchService implements Closeable {
                 apply(() -> client.updateJob(jobId, jobParameter))
             }
             catch (Exception e) {
-                log.warn "Unable to terminate Azure Batch job ${jobId} - Reason: ${e.message ?: e}"
+                if (e.message?.contains('Status code 409') && e.message?.contains('JobCompleted')) {
```

Comment:
Using a static exception type is may be better, see for example 

https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy#L873-L879